docs/build-with-rust.md

```diff
@@ -0,0 +1,354 @@
+
+    let sess = db::new_session_with_keyspace(&app.db_config).await?;
+
+    let (owner, pet, sensors) = random_data();
+    save_data(&sess, &owner, &pet, &sensors).await?;
+    run_sensor_data(&app, &sess, sensors).await?;
+
+    Ok(())
+}
+```
+
+The `app` object contains the command's arguments listed above.
+We then create a new session `sess` using `new_session_with_keyspace` function defined in `db/mod.rs`:
+
+```
+// db/mod.rs
+
+pub async fn new_session_with_keyspace(config: &Config) -> Result<Session> {
+    let session = new_session(config).await?;
+    session.use_keyspace(KEYSPACE, true).await?;
+    Ok(session)
+}
+```
+
+The `save_data` method connects to the datbase and saves random `owner`, `pet` and the `sensors` to the database using `insert_query` macro defined in `src/mod.rs`.
+
+```
+// sensor/main.rs
+
+async fn save_data(sess: &Session, owner: &Owner, pet: &Pet, sensors: &[Sensor]) -> Result<()> {
+    sess.query(insert_query!(Owner), owner).await?;
+    info!("New owner # {}", owner.owner_id);
+
+    sess.query(insert_query!(Pet), pet).await?;
+    info!("New pet # {}", pet.pet_id);
+
+    for sensor in sensors {
+        sess.query(insert_query!(Sensor), sensor).await?;
+    }
+
+    Ok(())
+}
+```
+
+The `run_sensor_data` generates random data and inserts it to the database every `buffer_interval`.
+
+```
+async fn run_sensor_data(cfg: &App, sess: &Session, sensors: Vec<Sensor>) -> Result<()> {
+    let measure: time::Duration = cfg.measure.into();
+    let buffer_interval: time::Duration = cfg.buffer_interval.into();
+
+    let mut last = Instant::now();
+    loop {
+        let mut measures = vec![];
+        while last.elapsed() < buffer_interval {
+            sleep(measure).await;
+
+            for sensor in &sensors {
+                let measure = read_sensor_data(sensor);
+                info!(
+                    "sensor # {} type {} new measure {} ts {}",
+                    sensor.sensor_id,
+                    sensor.r#type.as_str(),
+                    &measure.value,
+                    measure.ts.format_rfc3339(),
+                );
+
+                measures.push(measure);
+            }
+        }
+
+        last = last
+            + time::Duration::from_nanos(
+                (measure.as_nanos() * (last.elapsed().as_nanos() / measure.as_nanos())) as u64,
+            );
+
+        info!("Pushing data");
+
+        let batch = measures.iter().fold(Batch::default(), |mut batch, _| {
+            batch.append_statement(insert_query!(Measure));
+            batch
+        });
+
+        sess.batch(&batch, measures)
+            .await
+            .map_err(|err| error!("execute batch query {:?}", err))
+            .ok();
+    }
+}
+```
+
+### Server
+
+The server service is a REST API for tracking the pets’ health state. The service was built using [Rocket](https://rocket.rs) and allows users to query the database via http.
+
+Run the following commands to start the server:
+
+````
```

Comment:
```suggestion
```
```

Reply:
The extra "`" make everything from this point to the end of the page look like code
Please test with "make preview"